### PR TITLE
Auto save + disable upload while manuscript is uploading

### PR DESCRIPTION
**Pull request by norisalsaadie**
Closes https://github.com/libero/reviewer/issues/767 and https://github.com/libero/reviewer/issues/774

### CODE REVIEW

File: src/initial-submission/components/AuthorDetailsForm.tsx [hunk 1]
```diff
@@ -16,9 +16,10 @@ interface GetCurrentUser {
 
 interface Props {
     initialValues?: Submission;
+    setIsSaving?: any;
```
Comment (diversemix) [inline src/initial-submission/components/AuthorDetailsForm.tsx:19]: shouldn't this be bool?

Comment (norisalsaadie) [reply src/initial-submission/components/AuthorDetailsForm.tsx:19]: No, it's a function to update the state

Comment (diversemix) [reply src/initial-submission/components/AuthorDetailsForm.tsx:19]: Ah good ... main point being it shouldn't be "any" then :+1:

Comment (norisalsaadie) [reply src/initial-submission/components/AuthorDetailsForm.tsx]: Updated :)

Comment (norisalsaadie) [reply src/initial-submission/components/AuthorDetailsForm.tsx]: It's still a WIP, but struggling with the last use case without lifting state up. The use case is file uploads.

**Comment by hdrury1** [comment]
Assuming this will close https://github.com/libero/reviewer/issues/767

File: src/initial-submission/components/FileDetailsForm.tsx [hunk 2]
```diff
@@ -33,17 +33,47 @@ type UploadInProgress = {
 
 interface Props {
     initialValues?: Submission;
+    setIsSaving?: React.Dispatch<React.SetStateAction<boolean>>;
```
Comment (will-byrne) [inline src/initial-submission/components/FileDetailsForm.tsx:36]: why is this optional? We can guarantee that this is always there so if we make it required we can then remove `toggleSave`

File: src/initial-submission/components/SubmissionWizard.tsx [hunk 3]
```diff
@@ -83,12 +86,13 @@ const SubmissionWizard: React.FC<RouteComponentProps> = ({ history }: RouteCompo
...
             {getCurrentStepPathIndex() < stepConfig.length - 1 && (
                 <Button
+                    disabled={isSaving}
                     onClick={(): void => {
                         history.push(`/submit/${id}/${stepConfig[getCurrentStepPathIndex() + 1].id}`);
                     }}
                     type="primary"
                 >
-                    next
+                    {isSaving ? 'saving..' : 'next'}
```
Comment (diversemix) [inline src/initial-submission/components/SubmissionWizard.tsx:95]: only 2 dots after saving

File: src/ui/molecules/FileUpload.tsx [hunk 4]
```diff
@@ -107,6 +107,7 @@ interface Props {
         error?: UploadErrors;
         fileStored?: FileStored;
     };
+    // disabledUpload?: boolean;
```
Comment (diversemix) [inline src/ui/molecules/FileUpload.tsx:110]: remove commented code?

**Pull request merged**